### PR TITLE
Add diagnostic to detect unsupported or missing CCCL

### LINKED ISSUE #590: [ENHANCEMENT]: Emit diagnostic when detected CCCL version is not supported
### Is your feature request related to a problem? Please describe.

cuco's only dependency is CCCL and we make heavy use of new libcu++/CUB/Thrust features to improve our QoL as library devs.
Thus we require a fairly new CCCL version (IIRC currently it's v2.5).

Building cuco without any auxiliary dependency management like cmake/CPM can result in compilation errors, e.g., when cuco picks up the CCCLs version that shipped with an outdated CTK.

### Describe the solution you'd like

Check if the detected CCCL library version is sufficient and emit a compile-time diagnostic in case if it's not supported.
Rather than having to dive through some (sometimes obscure) compiler errors, the user is presented with a one-line error message.

### Describe alternatives you've considered

_No response_

### Additional context

Related issue: #588

### PR DESCRIPTION
**Pull request by sleeepyjack**
Closes #590

### CODE REVIEW

**Comment by sleeepyjack** [comment]
For the `__config` header, should we include it at the top of every public-facing header? My understanding is that this header should be parsed first in order to define some crucial symbols and check if the configuration is correct. Currently only a subset of our headers actually includes `__config`.

**Comment by PointKernel** [comment]
> For the `__config` header, should we include it at the top of every public-facing header? My understanding is that this header should be parsed first in order to define some crucial symbols and check if the configuration is correct. Currently only a subset of our headers actually includes `__config`.

Not necessary IMO. The build will fail anyway.

**Comment by sleeepyjack** [comment]
> The build will fail anyway.

True. But IIUC if `__config` is the first header that's parsed then we will see the diagnostic being triggered instead of an obscure compilation error somewhere else. Not 100% I understand this right.

**Comment by PointKernel** [comment]
> > The build will fail anyway.
> 
> True. But IIUC if `__config` is the first header that's parsed then we will see the diagnostic being triggered instead of an obscure compilation error somewhere else. Not 100% I understand this right.

Not worth the effort IMO. cuco users will inevitably encounter template build errors, so prioritizing the clarity of config errors isn't a major concern.

File: include/cuco/detail/__config [hunk 1]
```diff
@@ -17,6 +17,7 @@
 #pragma once
 
 #include <nv/target>
+#include <cuda/std/version>
```
Comment (PointKernel) [inline include/cuco/detail/__config:20]: question: is it possible that this header doesn't exist?

**Review by PointKernel — APPROVED** [review]
A small question but I don't need to review it again.

File: include/cuco/detail/__config (hunk 1 above)
Comment (sleeepyjack) [reply include/cuco/detail/__config:20]: Yeah, that was also my concern. Kind of an hen-egg problem so I tested it in godbold. It seems like the diagnostic is always triggered (first), even if the header doesn't exist. I still need to wrap my head around what's exactly going on but I think this PR is good to merge as of now.

**Pull request merged**